File: Paradigmas/Atv-07_07_jogo_adivinhacao/python/jogo.py
"""
Atividade - Paradigmas de Programacao
Jogo de Adivinhacao - Programacao estruturada
O computador sorteia um numero e o jogador tenta adivinhar.
"""

import random

MINIMO = 1
MAXIMO = 100
MAX_TENTATIVAS = 7


def ler_palpite() -> int:
    """Le um palpite valido do jogador (inteiro dentro do intervalo)."""
    while True:
        try:
            palpite = int(input(f"Seu palpite ({MINIMO}-{MAXIMO}): "))
        except ValueError:
            print("Digite um numero inteiro valido.")
            continue

        if palpite < MINIMO or palpite > MAXIMO:
            print(f"O numero deve estar entre {MINIMO} e {MAXIMO}.")
            continue

        return palpite


def jogar():
    numero_secreto = random.randint(MINIMO, MAXIMO)
    tentativas = 0

    print("=" * 50)
    print("   JOGO DE ADIVINHACAO")
    print(f"   Adivinhe o numero entre {MINIMO} e {MAXIMO}.")
    print(f"   Voce tem {MAX_TENTATIVAS} tentativas.")
    print("=" * 50)

    while tentativas < MAX_TENTATIVAS:
        palpite = ler_palpite()
        tentativas += 1

        if palpite == numero_secreto:
            print(f"\nParabens! Voce acertou em {tentativas} tentativa(s).")
            return
        elif palpite < numero_secreto:
            print("O numero secreto e MAIOR.")
        else:
            print("O numero secreto e MENOR.")

        restantes = MAX_TENTATIVAS - tentativas
        print(f"Tentativas restantes: {restantes}\n")

    print(f"\nVoce perdeu! O numero secreto era {numero_secreto}.")


if __name__ == "__main__":
    jogar()

File: Paradigmas/Atv-07_07_jogo_adivinhacao/javascript/jogo.js
/*
 * Atividade - Paradigmas de Programacao
 * Jogo de Adivinhacao - JavaScript (Node.js)
 * O computador sorteia um numero e o jogador tenta adivinhar.
 */

const readline = require("readline");

const MINIMO = 1;
const MAXIMO = 100;
const MAX_TENTATIVAS = 7;

// Sorteia um inteiro entre MINIMO e MAXIMO (inclusive)
const numeroSecreto = Math.floor(Math.random() * (MAXIMO - MINIMO + 1)) + MINIMO;
let tentativas = 0;

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

console.log("==================================================");
console.log("   JOGO DE ADIVINHACAO");
console.log(`   Adivinhe o numero entre ${MINIMO} e ${MAXIMO}.`);
console.log(`   Voce tem ${MAX_TENTATIVAS} tentativas.`);
console.log("==================================================");

function perguntar() {
    rl.question(`Seu palpite (${MINIMO}-${MAXIMO}): `, (entrada) => {
        const palpite = parseInt(entrada, 10);

        // Validacao da entrada
        if (Number.isNaN(palpite) || palpite < MINIMO || palpite > MAXIMO) {
            console.log(`Digite um numero inteiro entre ${MINIMO} e ${MAXIMO}.`);
            return perguntar();
        }

        tentativas++;

        if (palpite === numeroSecreto) {
            console.log(`\nParabens! Voce acertou em ${tentativas} tentativa(s).`);
            return rl.close();
        } else if (palpite < numeroSecreto) {
            console.log("O numero secreto e MAIOR.");
        } else {
            console.log("O numero secreto e MENOR.");
        }

        const restantes = MAX_TENTATIVAS - tentativas;
        if (restantes === 0) {
            console.log(`\nVoce perdeu! O numero secreto era ${numeroSecreto}.`);
            return rl.close();
        }

        console.log(`Tentativas restantes: ${restantes}\n`);
        perguntar();
    });
}

perguntar();
